src/Shell32.Desktop/Shell32+SHITEMID.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+
+namespace PInvoke
+{
+    using System.Runtime.InteropServices;
+
+    /// <content>
+    /// Contains the <see cref="SHITEMID"/> nested type.
+    /// </content>
+    public partial class Shell32
+    {
+        /// <summary>
+        /// Defines an item identifier.
+        /// </summary>
+        /// <remarks>Used by <see cref="ITEMIDLIST"/></remarks>
+        [StructLayout(LayoutKind.Sequential)]
+        public struct SHITEMID
+        {
+            /// <summary>
+            /// The size of identifier, in bytes, including cb itself.
+            /// </summary>
+            ushort cb;
+
+            /// <summary>
+            /// A variable-length item identifier.
+            /// </summary>
+            public byte[] abID;
```

Comment:
+1 to @vbfox's point of it breaking pinnable structs to use `byte[]`
But perhaps more importantly, I don't think your use of `byte[]` works here. That will result in an 8-byte struct, where the last 4 bytes are a pointer to the actual array. But my reading of the MSDN doc for this structure is that the array itself is inline with the struct itself. Basically this struct defines the structure of an array that has a leading `ushort` to specify its length. We have a *few* examples of inline arrays in structs already in this library, although I'm not sure where they are at. The fact that this one is variable length might be a new twist though. 
Is this why you added the `[MarshalAs(Struct)]` attribute on your use of this structure? Perhaps that does the magic you need, but it's unfortunate that it makes both structures un-pinnable, and requires special knowledge of this struct in every use of it.